src/control/server/ctl_storage_rpc.go

```diff
@@ -855,7 +855,8 @@ func formatNvme(ctx context.Context, req formatNvmeReq, resp *ctlpb.StorageForma
 		}
 
 		// SCM formatted correctly on this instance, format NVMe
-		cResults := formatEngineBdevs(engine, ctrlrs)
+		cResults := formatEngineBdevs(engine.(*EngineInstance), ctrlrs)
```

Comment:
Do we have reasons to be positive about this type underlying the interface? I know this is the type we actually have at runtime, but IMO we ought to check if the conversion worked, and return an error (or skip?) if not.

Reply:
EngineInstance is the only type that satisfies the Engine interface I think this is okay

Reply:
I do think the type ought to be checked anyway -- but I'm okay with it being addressed in a follow-on PR.

Reply:
PR-13463 is based on this PR branch and will need re-basing once this lands, I will add the check to that. Thanks Kris